File: tests/algorithms/x/Java/physics/lens_formulae.java
public class Main {

    static double focal_length_of_lens(double object_distance_from_lens, double image_distance_from_lens) {
        if ((double)(object_distance_from_lens) == (double)(0.0) || (double)(image_distance_from_lens) == (double)(0.0)) {
            throw new RuntimeException(String.valueOf("Invalid inputs. Enter non zero values with respect to the sign convention."));
        }
        return (double)((double)(1.0) / (double)(((double)(((double)(1.0) / (double)(image_distance_from_lens))) - (double)(((double)(1.0) / (double)(object_distance_from_lens))))));
    }

    static double object_distance(double focal_length_of_lens, double image_distance_from_lens) {
        if ((double)(image_distance_from_lens) == (double)(0.0) || (double)(focal_length_of_lens) == (double)(0.0)) {
            throw new RuntimeException(String.valueOf("Invalid inputs. Enter non zero values with respect to the sign convention."));
        }
        return (double)((double)(1.0) / (double)(((double)(((double)(1.0) / (double)(image_distance_from_lens))) - (double)(((double)(1.0) / (double)(focal_length_of_lens))))));
    }

    static double image_distance(double focal_length_of_lens, double object_distance_from_lens) {
        if ((double)(object_distance_from_lens) == (double)(0.0) || (double)(focal_length_of_lens) == (double)(0.0)) {
            throw new RuntimeException(String.valueOf("Invalid inputs. Enter non zero values with respect to the sign convention."));
        }
        return (double)((double)(1.0) / (double)(((double)(((double)(1.0) / (double)(object_distance_from_lens))) + (double)(((double)(1.0) / (double)(focal_length_of_lens))))));
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            System.out.println(_p(focal_length_of_lens((double)(10.0), (double)(4.0))));
            System.out.println(_p(focal_length_of_lens((double)(2.7), (double)(5.8))));
            System.out.println(_p(object_distance((double)(10.0), (double)(40.0))));
            System.out.println(_p(object_distance((double)(6.2), (double)(1.5))));
            System.out.println(_p(image_distance((double)(50.0), (double)(40.0))));
            System.out.println(_p(image_distance((double)(5.3), (double)(7.9))));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{\"duration_us\": " + _benchDuration + ", \"memory_bytes\": " + _benchMemory + ", \"name\": \"main\"}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        if (v instanceof java.util.Map<?, ?>) {
            StringBuilder sb = new StringBuilder("{");
            boolean first = true;
            for (java.util.Map.Entry<?, ?> e : ((java.util.Map<?, ?>) v).entrySet()) {
                if (!first) sb.append(", ");
                sb.append(_p(e.getKey()));
                sb.append("=");
                sb.append(_p(e.getValue()));
                first = false;
            }
            sb.append("}");
            return sb.toString();
        }
        if (v instanceof java.util.List<?>) {
            StringBuilder sb = new StringBuilder("[");
            boolean first = true;
            for (Object e : (java.util.List<?>) v) {
                if (!first) sb.append(", ");
                sb.append(_p(e));
                first = false;
            }
            sb.append("]");
            return sb.toString();
        }
        if (v instanceof Double || v instanceof Float) {
            double d = ((Number) v).doubleValue();
            return String.valueOf(d);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/Kotlin/physics/lens_formulae.kt
fun panic(msg: String): Nothing { throw RuntimeException(msg) }

var _nowSeed = 0L
var _nowSeeded = false
fun _now(): Long {
    if (!_nowSeeded) {
        System.getenv("MOCHI_NOW_SEED")?.toLongOrNull()?.let {
            _nowSeed = it
            _nowSeeded = true
        }
    }
    return if (_nowSeeded) {
        _nowSeed = (_nowSeed * 1664525 + 1013904223) % 2147483647
        kotlin.math.abs(_nowSeed)
    } else {
        kotlin.math.abs(System.nanoTime())
    }
}

fun toJson(v: Any?): String = when (v) {
    null -> "null"
    is String -> "\"" + v.replace("\"", "\\\"") + "\""
    is Boolean, is Number -> v.toString()
    is Map<*, *> -> v.entries.joinToString(prefix = "{", postfix = "}") { toJson(it.key.toString()) + ":" + toJson(it.value) }
    is Iterable<*> -> v.joinToString(prefix = "[", postfix = "]") { toJson(it) }
    else -> toJson(v.toString())
}

fun focal_length_of_lens(object_distance_from_lens: Double, image_distance_from_lens: Double): Double {
    if ((object_distance_from_lens == 0.0) || (image_distance_from_lens == 0.0)) {
        panic("Invalid inputs. Enter non zero values with respect to the sign convention.")
    }
    return 1.0 / ((1.0 / image_distance_from_lens) - (1.0 / object_distance_from_lens))
}

fun object_distance(focal_length_of_lens: Double, image_distance_from_lens: Double): Double {
    if ((image_distance_from_lens == 0.0) || (focal_length_of_lens == 0.0)) {
        panic("Invalid inputs. Enter non zero values with respect to the sign convention.")
    }
    return 1.0 / ((1.0 / image_distance_from_lens) - (1.0 / focal_length_of_lens))
}

fun image_distance(focal_length_of_lens: Double, object_distance_from_lens: Double): Double {
    if ((object_distance_from_lens == 0.0) || (focal_length_of_lens == 0.0)) {
        panic("Invalid inputs. Enter non zero values with respect to the sign convention.")
    }
    return 1.0 / ((1.0 / object_distance_from_lens) + (1.0 / focal_length_of_lens))
}

fun main() {
    run {
        System.gc()
        val _startMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
        val _start = _now()
        println(focal_length_of_lens(10.0, 4.0).toString())
        println(focal_length_of_lens(2.7, 5.8).toString())
        println(object_distance(10.0, 40.0).toString())
        println(object_distance(6.2, 1.5).toString())
        println(image_distance(50.0, 40.0).toString())
        println(image_distance(5.3, 7.9).toString())
        System.gc()
        val _end = _now()
        val _endMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
        val _durationUs = (_end - _start) / 1000
        val _memDiff = kotlin.math.abs(_endMem - _startMem)
        val _res = mapOf("duration_us" to _durationUs, "memory_bytes" to _memDiff, "name" to "main")
        println(toJson(_res))
    }
}
